copy/copy.go

```diff
@@ -126,6 +127,17 @@ type Options struct {
 	// Download layer contents with "nondistributable" media types ("foreign" layers) and translate the layer media type
 	// to not indicate "nondistributable".
 	DownloadForeignLayers bool
+
+	// Contains slice of OptionCompressionVariant, where copy will ensure that ImageListSelection
+	// images in the manifest list will exist with compression specificed in OptionCompressionVariant
+	// while performing copy. Invalid when copying a non-multi-architecture image. That will probably
+	// change in the future.
+	EnsureCompressionVariantsExist []OptionCompressionVariant
```

Comment:
Another reminder: `copy.Image` should fail on the non-`copyMultipleImages` paths if this is not set.

Reply:
If i got this right i think you meant `copy.Image should fail on the non-copyMultipleImages paths if this is set`, added this in a new commit.

Reply:
Yes, my mistake.